genshin/client/components/geetest/client.py

```diff
@@ -69,6 +69,54 @@ async def _web_login(
 
         return cookies
 
+    async def _cn_login_by_password(
+        self,
+        account: str,
+        password: str,
+        *,
+        geetest: typing.Optional[typing.Dict[str, typing.Any]] = None,
+    ) -> typing.Dict[str, typing.Any]:
+        """
+        Login with account and password using Miyoushe loginByPassword endpoint.
+
+        Returns data from aigis header or cookies.
+        """
+        headers = {
+            **geetest_utility.CN_LOGIN_HEADERS,
+            "ds": ds_utility.generate_dynamic_secret(constants.DS_SALT["cn_signin"]),
+        }
+        if geetest:
+            mmt_data = geetest["data"]
+            session_id = geetest["session_id"]
+            headers["x-rpc-aigis"] = geetest_utility.get_aigis_header(session_id, mmt_data)
+
+        payload = {
+            "account": geetest_utility.encrypt_geetest_credentials(account, self._region),
+            "password": geetest_utility.encrypt_geetest_credentials(password, self._region),
+        }
+
+        async with aiohttp.ClientSession() as session:
+            async with session.post(
+                routes.CN_LOGIN_URL.get_url(),
+                json=payload,
+                headers=headers,
+            ) as r:
+                data = await r.json()
+
+        if data["retcode"] == -3102:
+            # Captcha triggered
+            aigis = json.loads(r.headers["x-rpc-aigis"])
+            aigis["data"] = json.loads(aigis["data"])
+            return aigis
+
+        if not data["data"]:
+            errors.raise_for_retcode(data)
+
+        cookies = {cookie.key: cookie.value for cookie in r.cookies.values()}
+
+        self.set_cookies(cookies)
```

Comment:
`self.set_cookies(r.cookies)` should work I think?

Reply:
`r.cookies` is a `http.SimpleCookie` object, because the method wants to return cookies as dictionary, I decided to convert it earlier and use the same variable for both `set_cookies` and the return statement.